### ISSUE TITLE
Patterns ASV Table Counts

### ISSUE THREAD

**Issue by jsaw5** [benjjneb/dada2#549]
Hi there,

We came across some odd patterns in the counts in the ASV table (screenshot). It seems odd that: **(1)** I have so many repeat counts. **(2)** They are arranged in a pattern. The samples were numbered randomly, so we wouldn’t expect such orderly data. **(3)** The ASVs with high abundances don’t look suspicious.

<img width="1498" alt="screenshot01" src="https://user-images.githubusercontent.com/22158612/44601741-1b0aca80-a7a3-11e8-98d7-d56ff8c9ffa8.png">

Have you seen this before and should we be concerned about it? We checked on another 16S run and see the same ordering, though not as clean as this.  Also, note from the screenshot, where I have 61 counts of various ASVs, the counts in excel column ALR sum to 61.

This dataset is made of 18 tooth plaque samples and 15 kidney stone samples, **where the kidney stone samples are extremely low in abundance and the tooth plaque are extremely high**. Would making one error model out of two such communities be a potential problem, and possibly explain why the pattern is especially striking here? Would it be worth processing them separately?

Thanks a lot for your help!

Jessica

**Comment by cjfields** [comment]
@benjjneb Just to add for context: for @jsaw5 's work we're using a [workflow-based version](https://github.com/HPCBio/16S-rDNA-dada2-pipeline) for running on our local cluster, co-developed with H3ABioNet researchers.  

1.  I'm wondering whether the pattern above is partly due to how the sequence table is created (by default a sort is run by `colSums`):

https://github.com/benjjneb/dada2/blob/master/R/multiSample.R#L44

The snapshot above has samples with a colsum of 61 and then 60 (left to right), which explains the column ordering (I also confirmed the sort order with the full data set and with another run).  What I'm not sure about is whether (in the case of ties like the above) the sort order for the ties is then done by row (from top to bottom to bottom and left to right), which then creates the pattern.  

2. We're also unsure whether the lower-count samples @jsaw5 mentioned should be processed separately from the more complex plaque samples; these were combined in a single sequencing run and are testing whether doing this will affect counts.  We're testing this and will keep you posted, but any input would be greatly appreciated.

**Comment by benjjneb** [comment]
> I'm wondering whether the pattern above is partly due to how the sequence table is created (by default a sort is run by colSums

Yes, that probably is what is going on. Each of those columns has an ASV with the same total abundance and that is present in just one sample. The default R radix sort is "stable", so ties remain in the same order as the input order. Together this produces the pattern seen. So, the pattern itself is not a cause for concern, although it is worth checking how much of the data is explained by single-sample ASVs.

> We're also unsure whether the lower-count samples @jsaw5 mentioned should be processed separately from the more complex plaque samples; these were combined in a single sequencing run and are testing whether doing this will affect counts. We're testing this and will keep you posted, but any input would be greatly appreciated.

Were identical PCR protocols used in each case? That is another important source of error, as the error model is being fit to the sequencing+PCR error rates. If a different PCR protocol was used for the lo biomass samples, they should be run separately, so an appropriate error model is built for that PCR protocol. But if all processing was identical, then combining the low and high-biomass samples for the DADA2 part is fine.
